### PR TITLE
Added a guide to removing large files from repo history using bfg

### PR DESCRIPTION
**Pull request by dustinfreeman**
This guide requested in #207, and based on my own experience described in greater detail in this [blog post](https://dustinfreeman.org/blog/rip-lfs/)

I have tested this empirically in my own huge private game development repo as mentioned in the blog post, and just re-tested this behaviour with bfg-1.13.0 and git 2.19.1. I have to admit I don't have a deep understanding of git's backend, and have never even used git reflog, so a review of what I'm doing here is welcome.

### CODE REVIEW

**Comment by RichardLitt** [comment]
This is a good addition. The only thing I would add is a note that you may need to have Java working on your system before doing this.

Is there any chance you can explain how to do it with git filter-branch, too?

File: README.md [hunk 1]
```diff
@@ -379,6 +381,8 @@ With bfg the files that exist on your latest commit will not be affected. For ex
 
 Note, if you renamed your file as part of a commit, e.g. if it started as `LargeFileFirstName.mp4` and a commit changed it to `LargeFileSecondName.mp4`, running `java -jar ~/Downloads/bfg.jar --delete-files LargeFileSecondName.mp4` will not remove it from git history. Either run the `--delete-files` command with both filenames, or with a matching pattern. As explained above, any files present in the repo on your latest commit will be safe.
 
+#### Pushing your chnaged repo history
```
Comment (RichardLitt) [inline README.md]: Spelling

Comment (dustinfreeman) [reply README.md]: Darn, sorry, in the middle of work an inadverdently updated the pull request. Will notify here once it is completed.

**Comment by dustinfreeman** [comment]
Currently working on a git-filter-branch implementation, and testing thoroughly.

**Comment by dustinfreeman** [comment]
As I compare my git-filter-branch alternative to the [sensitive data removal technique above](https://github.com/k88hudson/git-flight-rules/tree/remove-big-files#i-accidentally-committed-and-pushed-files-containing-sensitive-data), I notice that that subsection only removes the sensitive data from the latest commit, not all history.

I'm going to finish this pull request with git-filter-branch instructions, but then the sensitive and large file removal sections should likely be merged.

**Comment by dustinfreeman** [comment]
This is done, but pending acceptance of #269

**Comment by dustinfreeman** [comment]
@deiga I've updated this PR on my fork with the latest from upstream master, including your #273 fix, but still have the old Travis CI error: 
```
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! git-flight-rules@1.0.0 diff: `test -z "`git diff -- README.md`" && test -z "`git diff -- README_*.md`"`
npm ERR! Exit status 1
```

I'm baffled. Is this TOC broken again? Is there some other way to read this error? Assistance appreciated.

**Comment by RichardLitt** [comment]
Yeah... I really don't care that much about the Travis error, if it is blocking us from having good content in here at all. I wouldn't worry about it. I'm going to merge this, because I think it's a useful addition. Good work. :)

**Pull request merged**